src/compiler.rs

```diff
@@ -35,9 +35,8 @@ pub const F32_TYPE_ID: usize = 11;
 pub const F64_TYPE_ID: usize = 12;
 pub const CCHAR_TYPE_ID: usize = 13;
 pub const CINT_TYPE_ID: usize = 14;
+// Note: keep USIZE_TYPE_ID last as it is how we know how many slots to pre-fill
 pub const USIZE_TYPE_ID: usize = 15;
```

Comment:
So... could we replace this with `pub const LAST_BUILTIN_TYPE_ID: usize = 15;` ?

Reply:
We could, but FOO_TYPE_IDs are treated as a valid unique type id, your suggestion kinda aliases the usize one
I don't mind either

Reply:
Dammit, wrong account _again_?

Reply:
Done.